resources/java/functions/LambdaExpression.java

```diff
@@ -0,0 +1,21 @@
+import java.util.function.Function;
+public class SampleFunctions {
+    public static void executeOperation(int value, MyFunctionalInterface operation) {
+        int result = operation.performOperation(value);
+        System.out.println("Operation Result: " + result);
+    }
+    public static int square(int num) {
+        Function<Integer, Integer> squareFunction = n -> n * n;
+        return squareFunction.apply(num);
+    }
+    public static void main(String[] args) {
+        executeOperation(4, (x) -> x * 2); // Lambda expression for doubling the value
+
+        int squaredResult = square(3);
+        System.out.println("Square Result: " + squaredResult);
+    }
+    @FunctionalInterface
+    interface MyFunctionalInterface {
+        int performOperation(int value);
```

Comment:
That also counts as function definition here, mh... 
For C++, we (still) do not count function declarations without actual implementation at that place as a function currently and it definitely makes sense there. So it would again be a deviation between the metric result between different programming languages to count such pure function declarations/interfaces here. But there does not seem to be a way to change this behaviour for Java with the config file only, again :(

Reply:
I had opened issue #98 to discuss that. Based on your suggestion, I guess that we should change the behaviour for C++ instead within the scope of that new issue.